### PR TITLE
Add ability to check address ranges

### PR DESCRIPTION
**Pull request by drallgood**
This adds the ability to use CIDR formatted address ranges like `10.0.1.1/24` (10.0.1.1-10.0.1.255) or `10.0.0.1/8` (10.0.0.1-10.255.255.255)

### CODE REVIEW

**Comment by chamini2** [comment]
When starting the example server, http://localhost:3000/authenticated throws a 500. Can you check that out?

**Comment by drallgood** [comment]
@chamini2 Good catch! I think I messed up the file when cherry picking the changes from another file.
Should be fixed now.

File: lib/index.js [hunk 1]
```diff
@@ -17,8 +17,15 @@ function ipWhitelistScheme(server, whitelisted) {
...
       const list = whitelisted instanceof Array ? whitelisted : [whitelisted]
-      if (list.some(ip => ipIsEqual(ip, remoteAddress))) {
-        return h.authenticated({ credentials: remoteAddress })
+      if (list.some(ip => {
+          if ((ipLib.isV4Format(ip) || ipLib.isV6Format(ip)) && ipLib.isEqual(ip, remoteAddress)) {
+              return true;
+          } else if (ip.includes("/") && ipLib.cidrSubnet(ip).contains(remoteAddress)) {
+              return true
+          }
+          return false
```
Comment (chamini2) [inline lib/index.js]: Suggestion:
```js
const regularIP = ipLib.isV4Format(ip) || ipLib.isV6Format(ip);
const cidrIP = ip.includes("/") && ipLib.cidrSubnet(ip).contains(remoteAddress);

return regularIP && ipLib.isEqual(ip, remoteAddress) || cidrIP;
```

**Review by chamini2 — CHANGES_REQUESTED** [review]

**Comment by drallgood** [comment]
@chamini2 I made the requested changes

**Review by chamini2 — APPROVED** [review]

**Comment by chamini2** [comment]
Thank you for your contribution!

**Pull request merged**